tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -1289,6 +1289,60 @@ uint32_t SPIRVEmitter::doConditionalOperator(const ConditionalOperator *expr) {
 }
 
 uint32_t
+SPIRVEmitter::processByteAddressBufferStore(const CXXMemberCallExpr *expr,
+                                            uint32_t numWords) {
+  const auto object = expr->getImplicitObjectArgument();
+  const auto type = object->getType();
+  const uint32_t objectId = doExpr(object);
+  assert(typeTranslator.isRWByteAddressBuffer(type));
+  assert(numWords == 1 || numWords == 2 || numWords == 3 || numWords == 4);
+  assert(expr->getNumArgs() == 2);
+  const Expr *addressExpr = expr->getArg(0);
+  const uint32_t valuesId = doExpr(expr->getArg(1));
+  const uint32_t byteAddress = doExpr(addressExpr);
+  const uint32_t addressTypeId =
+      typeTranslator.translateType(addressExpr->getType());
+
+  // Do a OpShiftRightLogical by 2 (divide by 4 to get aligned memory
+  // access). The AST always casts the address to unsinged integer, so shift
+  // by unsinged integer 2.
+  const uint32_t constUint2 = theBuilder.getConstantUint32(2);
+  const uint32_t storeAddress = theBuilder.createBinaryOp(
+      spv::Op::OpShiftRightLogical, addressTypeId, byteAddress, constUint2);
+
```

Comment:
I think that the code from the beginning of this function up to this point can be shared between `processByteAddressBufferLoad` and this one. Combining this two methods by providing an additional parameter in the method to indicate whether for load or store maybe? We'll have less code generated and less code to maintain.